mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/SavingsChargeDialogFragment.java

```diff
@@ -0,0 +1,245 @@
+    @Override
+    public void onDatePicked(String date) {
+        charge_due_date.setText(date);
+    }
+
+    private void inflateChargesSpinner() {
+        showProgress(true);
+        App.apiManager.getAllSavingsCharges(savingsAccountNumber,new Callback<Response>() {
+
+            @Override
+            public void success(final Response result, Response response) {
+                /* Activity is null - Fragment has been detached; no need to do anything. */
+                if (getActivity() == null) return;
+                Log.d(TAG, "");
+
+                final List<Charges> charges = new ArrayList<>();
+                // you can use this array to populate your spinner
+                final ArrayList<String> chargesNames = new ArrayList<String>();
+                //Try to get response body
+                BufferedReader reader = null;
+                StringBuilder sb = new StringBuilder();
+                try {
+                    reader = new BufferedReader(new InputStreamReader(result.getBody().in()));
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        sb.append(line);
+                    }
+                    JSONObject obj = new JSONObject(sb.toString());
+                    if (obj.has("chargeOptions")) {
+                        JSONArray chargesTypes = obj.getJSONArray("chargeOptions");
+                        for (int i = 0; i < chargesTypes.length(); i++) {
+                            JSONObject chargesObject = chargesTypes.getJSONObject(i);
+                            Charges charge = new Charges();
+                            charge.setId(chargesObject.optInt("id"));
+                            charge.setName(chargesObject.optString("name"));
+                            charges.add(charge);
+                            chargesNames.add(chargesObject.optString("name"));
+                            chargeNameIdHashMap.put(charge.getName(), charge.getId());
+                        }
+
+                    }
+                    String stringResult = sb.toString();
+                } catch (Exception e) {
+                    Log.e(TAG, "", e);
+                }
+                final ArrayAdapter<String> chargesAdapter = new ArrayAdapter<String>(getActivity(),
+                        android.R.layout.simple_spinner_item, chargesNames);
+                chargesAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+                sp_charge_name.setAdapter(chargesAdapter);
+                sp_charge_name.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+                    @Override
+                    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                        Id = chargeNameIdHashMap.get(chargesNames.get(i));
+                        Log.d("chargesoptionss" + chargesNames.get(i), String.valueOf(Id));
+                        if (Id != -1) {
```

Comment:
That block of code deals with loading the savings charges options into the spinner for selecting charge options using the spinner.

I tested using the domain of demo.openmf.org with the account of  John Kwame Doe (Client Id 201). I was able to load the four charge options using JSON and selected one charge from the spinner.